5.0.0: build check fails for TestKeeperTestSuite and TestIBCTestingSuite

Building on Arch Linux in a clean chroot, the following checks fail:

```plaintext
==> Starting check()...
go: downloading github.com/onsi/ginkgo/v2 v2.1.4
go: downloading github.com/onsi/gomega v1.19.0
go: downloading github.com/stretchr/objx v0.4.0
ok      github.com/Canto-Network/Canto/v2/app   0.279s
ok      github.com/Canto-Network/Canto/v2/app/ante      0.203s
ok      github.com/Canto-Network/Canto/v2/app/upgrades/v2       0.077s [no tests to run]
?       github.com/Canto-Network/Canto/v2/app/upgrades/v3       [no test files]
?       github.com/Canto-Network/Canto/v2/app/upgrades/v4       [no test files]
?       github.com/Canto-Network/Canto/v2/app/upgrades/v5       [no test files]
?       github.com/Canto-Network/Canto/v2/client/docs/statik    [no test files]
ok      github.com/Canto-Network/Canto/v2/cmd/cantod    0.173s
?       github.com/Canto-Network/Canto/v2/cmd/config    [no test files]
?       github.com/Canto-Network/Canto/v2/contracts     [no test files]
?       github.com/Canto-Network/Canto/v2/crypto/keyring        [no test files]
ok      github.com/Canto-Network/Canto/v2/ibc   0.182s
?       github.com/Canto-Network/Canto/v2/ibc/testing   [no test files]
?       github.com/Canto-Network/Canto/v2/testutil      [no test files]
?       github.com/Canto-Network/Canto/v2/testutil/network      [no test files]
ok      github.com/Canto-Network/Canto/v2/types 0.105s
?       github.com/Canto-Network/Canto/v2/version       [no test files]
?       github.com/Canto-Network/Canto/v2/x/csr [no test files]
?       github.com/Canto-Network/Canto/v2/x/csr/client/cli      [no test files]
ok      github.com/Canto-Network/Canto/v2/x/csr/keeper  0.678s
ok      github.com/Canto-Network/Canto/v2/x/csr/types   0.174s
ok      github.com/Canto-Network/Canto/v2/x/epochs      0.324s
?       github.com/Canto-Network/Canto/v2/x/epochs/client/cli   [no test files]
--- FAIL: TestKeeperTestSuite (0.08s)
    --- FAIL: TestKeeperTestSuite/TestEpochInfo (0.03s)
        --- FAIL: TestKeeperTestSuite/TestEpochInfo/Case_set_epoch_info (0.02s)
panic: EVM::DeployTurnstile error deploying Turnstile: EVM::DeployContract error deploying contract: failed to load evm config: failed to obtain coinbase address: failed to retrieve validator from block proposer address : validator does not exist: There was an error deploying a contract from the CSR module: There was an error deploying a contract from the CSR module [recovered]
        panic: EVM::DeployTurnstile error deploying Turnstile: EVM::DeployContract error deploying contract: failed to load evm config: failed to obtain coinbase address: failed to retrieve validator from block proposer address : validator does not exist: There was an error deploying a contract from the CSR module: There was an error deploying a contract from the CSR module

goroutine 387 [running]:
testing.tRunner.func1.2({0x55edd4f03ec0, 0xc0010f3c80})
        testing/testing.go:1396 +0x24e
testing.tRunner.func1()
        testing/testing.go:1399 +0x39f
panic({0x55edd4f03ec0, 0xc0010f3c80})
        runtime/panic.go:884 +0x212
github.com/Canto-Network/Canto/v2/x/csr.AppModule.BeginBlock({{{0x0, 0x0}}, {{0x55edd50c18e0, 0xc0010c9810}, {0x55edd50e3f80, 0xc00107bc40}, {{0x55edd50e3f80, 0xc00107bc40}, 0xc000ccfe90, {0x55edd50c18e0, ...}, ...}, ...}, ...}, ...)
        github.com/Canto-Network/Canto/v2/x/csr/module.go:176 +0x213
github.com/cosmos/cosmos-sdk/types/module.(*Manager).BeginBlock(_, {{0x55edd50d6a60, 0xc000132000}, {0x55edd50e56e0, 0xc00117f8c0}, {{0xb, 0x0}, {0x55edd42307d9, 0xc}, 0x2, ...}, ...}, ...)
        github.com/cosmos/cosmos-sdk@v0.45.9/types/module/module.go:491 +0x3bb
github.com/Canto-Network/Canto/v2/app.(*Canto).BeginBlocker(_, {{0x55edd50d6a60, 0xc000132000}, {0x55edd50e56e0, 0xc00117f8c0}, {{0xb, 0x0}, {0x55edd42307d9, 0xc}, 0x2, ...}, ...}, ...)
        github.com/Canto-Network/Canto/v2/app/app.go:832 +0x85
github.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).BeginBlock(_, {{0x0, 0x0, 0x0}, {{0xb, 0x0}, {0x55edd42307d9, 0xc}, 0x2, {0x1f905d11, ...}, ...}, ...})
        github.com/cosmos/cosmos-sdk@v0.45.9/baseapp/abci.go:177 +0x97b
github.com/Canto-Network/Canto/v2/x/epochs/keeper_test.(*KeeperTestSuite).CommitAfter(0xc0000df340, 0xc000132000?)
        github.com/Canto-Network/Canto/v2/x/epochs/keeper/keeper_test.go:111 +0x1f8
github.com/Canto-Network/Canto/v2/x/epochs/keeper_test.(*KeeperTestSuite).Commit(...)
        github.com/Canto-Network/Canto/v2/x/epochs/keeper/keeper_test.go:103
github.com/Canto-Network/Canto/v2/x/epochs/keeper_test.(*KeeperTestSuite).TestEpochInfo.func2()
        github.com/Canto-Network/Canto/v2/x/epochs/keeper/grpc_query_test.go:101 +0x3f8
github.com/Canto-Network/Canto/v2/x/epochs/keeper_test.(*KeeperTestSuite).TestEpochInfo.func3()
        github.com/Canto-Network/Canto/v2/x/epochs/keeper/grpc_query_test.go:120 +0x12f
github.com/stretchr/testify/suite.(*Suite).Run.func1(0x0?)
        github.com/stretchr/testify@v1.8.0/suite/suite.go:91 +0x36
testing.tRunner(0xc0010d1ba0, 0xc000d2e120)
        testing/testing.go:1446 +0x10b
created by testing.(*T).Run
        testing/testing.go:1493 +0x35f
FAIL    github.com/Canto-Network/Canto/v2/x/epochs/keeper       0.219s
ok      github.com/Canto-Network/Canto/v2/x/epochs/types        0.118s
ok      github.com/Canto-Network/Canto/v2/x/erc20       0.164s
?       github.com/Canto-Network/Canto/v2/x/erc20/client        [no test files]
?       github.com/Canto-Network/Canto/v2/x/erc20/client/cli    [no test files]
?       github.com/Canto-Network/Canto/v2/x/erc20/client/rest   [no test files]
ok      github.com/Canto-Network/Canto/v2/x/erc20/keeper        5.095s
ok      github.com/Canto-Network/Canto/v2/x/erc20/migrations/v2 0.178s
ok      github.com/Canto-Network/Canto/v2/x/erc20/types 0.102s
?       github.com/Canto-Network/Canto/v2/x/fees        [no test files]
?       github.com/Canto-Network/Canto/v2/x/fees/client/cli     [no test files]
Running Suite: Keeper Suite - /build/canto/src/canto/x/fees/keeper
==================================================================
Random Seed: 1674667676

Will run 23 of 23 specs
------------------------------
•! [PANICKED] [0.018 seconds]
Fee distribution: [BeforeAll]
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:60
  with fees param disabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:84
    should not allow new contract registrations
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:105

  Test Panicked
  In [BeforeAll] at: testing/testing.go:871

  <nil>

  Full Stack Trace
    testing.(*common).FailNow(0xc00120f860)
        testing/testing.go:871 +0x4c
    github.com/stretchr/testify/require.NotEmpty({0x55f7bf9a1568, 0xc00120f860}, {0x55f7bf6f6b00, 0x0}, {0x0, 0x0, 0x0})
        github.com/stretchr/testify@v1.8.0/require/require.go:1352 +0xb4
    github.com/stretchr/testify/require.(*Assertions).NotEmpty(0xc0014ba2b0, {0x55f7bf6f6b00, 0x0}, {0x0, 0x0, 0x0})
        github.com/stretchr/testify@v1.8.0/require/require_forward.go:1061 +0x87
    github.com/Canto-Network/Canto/v2/x/fees/keeper_test.deployContract(0xc000686ff0, {0x55f7beb934f3, 0x7c})
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:808 +0x519
    github.com/Canto-Network/Canto/v2/x/fees/keeper_test.glob..func1.1()
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:80 +0x8c7
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param disabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:84
    [It] should not distribute tx fees for previously registered contracts
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:126

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:126
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param disabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:84
    [It] should not allow fee information updates for previously registered contracts
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:136

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:136
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param disabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:84
    [It] should not allow cancellations of previously registered contracts
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:152

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:152
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Registering a contract for receiving tx fees
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:171
      with an empty withdrawal address
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:172
        [It] should be possible
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:181

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:181
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Registering a contract for receiving tx fees
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:171
      with an empty withdrawal address
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:172
        [It] should result in sending the tx fees to the deployer address
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:191

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:191
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Registering a contract for receiving tx fees
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:171
      with a withdrawal address equal to the deployer address
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:204
        [It] should be possible
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:205

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:205
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Registering a contract for receiving tx fees
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:171
      with a withdrawal address different than deployer
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:219
        [It] should be possible
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:230

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:230
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Registering a contract for receiving tx fees
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:171
      with a withdrawal address different than deployer
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:219
        [It] should send the fees to the withdraw address
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:240

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:240
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Interacting with a registered fee contract
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:254
      with a 50/50 validators-developers fee split
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:264
        [It] should transfer legacy tx fees to validators and contract developer evenly
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:271

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:271
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Interacting with a registered fee contract
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:254
      with a 50/50 validators-developers fee split
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:264
        [It] should transfer dynamic tx fees to validators and contract developer evenly
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:288

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:288
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Interacting with a registered fee contract
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:254
      with a 100/0 validators-developers fee split
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:311
        [It] should transfer all tx fees to validators
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:318

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:318
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Interacting with a registered fee contract
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:254
      with a 0/100 validators-developers fee split
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:341
        [It] should transfer all tx fees to developers
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:348

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:348
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Updating registered fee information
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:372
      with a withdraw address that is different from the deployer address
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:373
        [It] should update fee information successfully
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:391

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:391
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Updating registered fee information
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:372
      with a withdraw address that is different from the deployer address
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:373
        [It] should send tx fees to the new withdraw address
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:413

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:413
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Updating registered fee information
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:372
      with a withdraw address equal to the deployer's address
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:428
        [It] should not update fee information
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:444

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:444
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Updating registered fee information
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:372
      for a contract that was not registered
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:464
        [It] should fail
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:465

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:465
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Canceling a fee registration
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:488
      when the registered fee info exists
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:489
        [It] should be possible
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:505

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:505
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Canceling a fee registration
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:488
      when the registered fee info exists
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:489
        [It] should no longer distribute fees to the contract deployer
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:519

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:519
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Canceling a fee registration
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:488
      when the fee info does not exist
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:531
        [It] should not be possible
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:532

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:532
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Registering contracts created by a factory contract with CREATE opcode
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:549
      with one factory
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:550
        [It] should be possible
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:560

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:560
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Registering contracts created by a factory contract with CREATE opcode
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:549
      with one factory
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:550
        [It] should transfer legacy tx fees evenly to validator and deployer
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:578

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:578
------------------------------
S [SKIPPED] [0.000 seconds]
Fee distribution:
github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:42
  with fees param enabled
  github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:164
    Registering contracts created by a factory contract with CREATE opcode
    github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:549
      with one factory
      github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:550
        [It] should transfer dynamic tx fees evenly to validator and deployer
        github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:591

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/fees/keeper/integration_test.go:591
------------------------------

Ran 1 of 23 Specs in 0.020 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 22 Skipped
--- FAIL: TestKeeperTestSuite (0.75s)
    integration_test.go:808: 
                Error Trace:    /build/canto/src/canto/x/fees/keeper/integration_test.go:808
                                                        /build/canto/src/canto/x/fees/keeper/integration_test.go:80
                                                        /build/canto/src/canto/x/fees/keeper/suite.go:596
                                                        /build/canto/src/canto/x/fees/keeper/asm_amd64.s:1594
                Error:          Should NOT be empty, but was <nil>
                Test:           TestKeeperTestSuite
FAIL
FAIL    github.com/Canto-Network/Canto/v2/x/fees/keeper 0.923s
ok      github.com/Canto-Network/Canto/v2/x/fees/types  0.099s
ok      github.com/Canto-Network/Canto/v2/x/govshuttle  0.202s
?       github.com/Canto-Network/Canto/v2/x/govshuttle/client   [no test files]
?       github.com/Canto-Network/Canto/v2/x/govshuttle/client/cli       [no test files]
ok      github.com/Canto-Network/Canto/v2/x/govshuttle/keeper   0.121s
?       github.com/Canto-Network/Canto/v2/x/govshuttle/simulation       [no test files]
ok      github.com/Canto-Network/Canto/v2/x/govshuttle/types    0.096s
?       github.com/Canto-Network/Canto/v2/x/inflation   [no test files]
?       github.com/Canto-Network/Canto/v2/x/inflation/client/cli        [no test files]
--- FAIL: TestKeeperTestSuite (0.03s)
    --- FAIL: TestKeeperTestSuite/TestEpochMintProvision (0.02s)
        --- FAIL: TestKeeperTestSuite/TestEpochMintProvision/Case_set_epochMintProvision (0.01s)
panic: EVM::DeployTurnstile error deploying Turnstile: EVM::DeployContract error deploying contract: failed to load evm config: failed to obtain coinbase address: failed to retrieve validator from block proposer address : validator does not exist: There was an error deploying a contract from the CSR module: There was an error deploying a contract from the CSR module [recovered]
        panic: EVM::DeployTurnstile error deploying Turnstile: EVM::DeployContract error deploying contract: failed to load evm config: failed to obtain coinbase address: failed to retrieve validator from block proposer address : validator does not exist: There was an error deploying a contract from the CSR module: There was an error deploying a contract from the CSR module

goroutine 336 [running]:
testing.tRunner.func1.2({0x55bd723b40c0, 0xc00174c1a0})
        testing/testing.go:1396 +0x24e
testing.tRunner.func1()
        testing/testing.go:1399 +0x39f
panic({0x55bd723b40c0, 0xc00174c1a0})
        runtime/panic.go:884 +0x212
github.com/Canto-Network/Canto/v2/x/csr.AppModule.BeginBlock({{{0x0, 0x0}}, {{0x55bd72573060, 0xc0015e4050}, {0x55bd72595620, 0xc0015ac5c0}, {{0x55bd72595620, 0xc0015ac5c0}, 0xc000fc99a0, {0x55bd72573060, ...}, ...}, ...}, ...}, ...)
        github.com/Canto-Network/Canto/v2/x/csr/module.go:176 +0x213
github.com/cosmos/cosmos-sdk/types/module.(*Manager).BeginBlock(_, {{0x55bd72588280, 0xc0001b2000}, {0x55bd72596d80, 0xc001715900}, {{0xb, 0x0}, {0x55bd716da87c, 0xc}, 0x2, ...}, ...}, ...)
        github.com/cosmos/cosmos-sdk@v0.45.9/types/module/module.go:491 +0x3bb
github.com/Canto-Network/Canto/v2/app.(*Canto).BeginBlocker(_, {{0x55bd72588280, 0xc0001b2000}, {0x55bd72596d80, 0xc001715900}, {{0xb, 0x0}, {0x55bd716da87c, 0xc}, 0x2, ...}, ...}, ...)
        github.com/Canto-Network/Canto/v2/app/app.go:832 +0x85
github.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).BeginBlock(_, {{0x0, 0x0, 0x0}, {{0xb, 0x0}, {0x55bd716da87c, 0xc}, 0x2, {0x3b30c8e1, ...}, ...}, ...})
        github.com/cosmos/cosmos-sdk@v0.45.9/baseapp/abci.go:177 +0x97b
github.com/Canto-Network/Canto/v2/x/inflation/keeper_test.(*KeeperTestSuite).CommitAfter(0xc0000ea580, 0xc0015e4150?)
        github.com/Canto-Network/Canto/v2/x/inflation/keeper/keeper_test.go:112 +0x1f8
github.com/Canto-Network/Canto/v2/x/inflation/keeper_test.(*KeeperTestSuite).Commit(...)
        github.com/Canto-Network/Canto/v2/x/inflation/keeper/keeper_test.go:104
github.com/Canto-Network/Canto/v2/x/inflation/keeper_test.(*KeeperTestSuite).TestEpochMintProvision.func2()
        github.com/Canto-Network/Canto/v2/x/inflation/keeper/grpc_query_test.go:95 +0x16b
github.com/Canto-Network/Canto/v2/x/inflation/keeper_test.(*KeeperTestSuite).TestEpochMintProvision.func3()
        github.com/Canto-Network/Canto/v2/x/inflation/keeper/grpc_query_test.go:108 +0x12f
github.com/stretchr/testify/suite.(*Suite).Run.func1(0xc9?)
        github.com/stretchr/testify@v1.8.0/suite/suite.go:91 +0x36
testing.tRunner(0xc00158c1a0, 0xc001584ea0)
        testing/testing.go:1446 +0x10b
created by testing.(*T).Run
        testing/testing.go:1493 +0x35f
FAIL    github.com/Canto-Network/Canto/v2/x/inflation/keeper    0.129s
?       github.com/Canto-Network/Canto/v2/x/inflation/migrations/v2     [no test files]
ok      github.com/Canto-Network/Canto/v2/x/inflation/types     0.116s
ok      github.com/Canto-Network/Canto/v2/x/recovery    0.113s
?       github.com/Canto-Network/Canto/v2/x/recovery/client/cli [no test files]
Running Suite: Keeper Suite - /build/canto/src/canto/x/recovery/keeper
======================================================================
Random Seed: 1674667680

Will run 6 of 6 specs
------------------------------
•! [PANICKED] [0.010 seconds]
Recovery: Performing an IBC Transfer [BeforeEach]
github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:27
  from a non-authorized chain
  github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:31
    should transfer and not recover tokens
    github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:41

  Test Panicked
  In [BeforeEach] at: github.com/Canto-Network/Canto/v2/x/csr/module.go:176

  EVM::DeployTurnstile error deploying Turnstile: EVM::DeployContract error deploying contract: failed to load evm config: failed to obtain coinbase address: failed to retrieve validator from block proposer address : validator does not exist: There was an error deploying a contract from the CSR module: There was an error deploying a contract from the CSR module [Canto-Network/Canto/v2/x/csr/keeper/evm.go:27]

  Full Stack Trace
    github.com/Canto-Network/Canto/v2/x/csr.AppModule.BeginBlock({{{0x0, 0x0}}, {{0x55a4de9edd50, 0xc000210c20}, {0x55a4dea109a0, 0xc000e4e180}, {{0x55a4dea109a0, 0xc000e4e180}, 0xc000014088, {0x55a4de9edd50, ...}, ...}, ...}, ...}, ...)
        github.com/Canto-Network/Canto/v2/x/csr/module.go:176 +0x213
    github.com/cosmos/cosmos-sdk/types/module.(*Manager).BeginBlock(_, {{0x55a4dea030e0, 0xc0000560c8}, {0x55a4dea12420, 0xc00128c040}, {{0x0, 0x0}, {0xc000e863e0, 0xc}, 0x2, ...}, ...}, ...)
        github.com/cosmos/cosmos-sdk@v0.45.9/types/module/module.go:491 +0x3bb
    github.com/Canto-Network/Canto/v2/app.(*Canto).BeginBlocker(_, {{0x55a4dea030e0, 0xc0000560c8}, {0x55a4dea12420, 0xc00128c040}, {{0x0, 0x0}, {0xc000e863e0, 0xc}, 0x2, ...}, ...}, ...)
        github.com/Canto-Network/Canto/v2/app/app.go:832 +0x85
    github.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).BeginBlock(_, {{0x0, 0x0, 0x0}, {{0x0, 0x0}, {0xc000e863e0, 0xc}, 0x2, {0x0, ...}, ...}, ...})
        github.com/cosmos/cosmos-sdk@v0.45.9/baseapp/abci.go:177 +0x97b
    github.com/Canto-Network/Canto/v2/ibc/testing.SetupWithGenesisValSet(0x55a4dea0d7a8?, 0xc000e98510, {0xc0013e14d0?, 0x1, 0x1}, {0xc000e863e0, 0xc}, {0xc000e98cf0, 0x1, 0x1})
        github.com/Canto-Network/Canto/v2/ibc/testing/app.go:100 +0x19be
    github.com/Canto-Network/Canto/v2/ibc/testing.NewTestChain(0xc00016da00, 0xc000e98390, {0xc000e863e0, 0xc})
        github.com/Canto-Network/Canto/v2/ibc/testing/chain.go:72 +0x65a
    github.com/Canto-Network/Canto/v2/ibc/testing.NewCoordinator(0xc00016da00, 0x1, 0x2)
        github.com/Canto-Network/Canto/v2/ibc/testing/coordinator.go:25 +0x152
    github.com/Canto-Network/Canto/v2/x/recovery/keeper_test.(*IBCTestingSuite).SetupTest(0xc0003d9c70)
        github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_suite_test.go:53 +0x47
    github.com/Canto-Network/Canto/v2/x/recovery/keeper_test.glob..func1.1()
        github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:28 +0x25
------------------------------
S [SKIPPED] [0.000 seconds]
Recovery: Performing an IBC Transfer
github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:15
  from an authorized, non-EVM chain (e.g. Osmosis)
  github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:51
    to a different account on canto (sender != recipient)
    github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:52
      [It] should transfer and not recover tokens
      github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:60

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:60
------------------------------
S [SKIPPED] [0.000 seconds]
Recovery: Performing an IBC Transfer
github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:15
  from an authorized, non-EVM chain (e.g. Osmosis)
  github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:51
    to the sender's own eth_secp256k1 account on canto (sender == recipient)
    github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:70
      with disabled recovery parameter
      github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:78
        [It] should not transfer or recover tokens
        github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:85

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:85
------------------------------
S [SKIPPED] [0.000 seconds]
Recovery: Performing an IBC Transfer
github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:15
  from an authorized, non-EVM chain (e.g. Osmosis)
  github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:51
    to the sender's own eth_secp256k1 account on canto (sender == recipient)
    github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:70
      without a sender's claims record
      github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:95
        when recipient has no ibc vouchers that originated from other chains
        github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:96
          [It] should transfer and recover tokens
          github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:97

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:97
------------------------------
S [SKIPPED] [0.000 seconds]
Recovery: Performing an IBC Transfer
github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:15
  from an authorized, non-EVM chain (e.g. Osmosis)
  github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:51
    to the sender's own eth_secp256k1 account on canto (sender == recipient)
    github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:70
      without a sender's claims record
      github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:95
        when recipient has additional ibc vouchers that originated from other chains
        github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:132
          [It] should not recover tokens that originated from other chains
          github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:144

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:144
------------------------------
S [SKIPPED] [0.000 seconds]
Recovery: Performing an IBC Transfer
github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:15
  from an authorized, non-EVM chain (e.g. Osmosis)
  github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:51
    to the sender's own eth_secp256k1 account on canto (sender == recipient)
    github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:70
      without a sender's claims record
      github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:95
        when recipient has additional non-native ibc vouchers that originated from senders chains
        github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:195
          [It] should not recover tokens that originated from other chains
          github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:220

  Spec skipped because an earlier spec in an ordered container failed
  In [It] at: github.com/Canto-Network/Canto/v2/x/recovery/keeper/ibc_callbacks_integration_test.go:220
------------------------------

Ran 1 of 6 Specs in 0.010 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 5 Skipped
--- FAIL: TestIBCTestingSuite (0.01s)
    suite.go:207: warning: no tests to run
FAIL
FAIL    github.com/Canto-Network/Canto/v2/x/recovery/keeper     0.258s
ok      github.com/Canto-Network/Canto/v2/x/recovery/types      0.097s
?       github.com/Canto-Network/Canto/v2/x/vesting     [no test files]
?       github.com/Canto-Network/Canto/v2/x/vesting/client/cli  [no test files]
ok      github.com/Canto-Network/Canto/v2/x/vesting/keeper      1.027s
ok      github.com/Canto-Network/Canto/v2/x/vesting/types       0.071s
FAIL
```

## System Info

**OS:** `Linux cryptsus 6.1.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Wed, 18 Jan 2023 19:54:38 +0000 x86_64 GNU/Linux`
**go (version):** `1.19.5`

--- a/x/csr/keeper/integration_test.go
+++ b/x/csr/keeper/integration_test.go
@@ -112,6 +112,7 @@ var _ = Describe("CSR Distribution : ", Ordered, func() {
 			s.Require().False(found)
 		})
 		It("it should register a smart contract (non-factory deployed)", func() {
+
 			// Deploys the contract directly to the EVM state (does not hit the postTxProcessing hooks)
 			contractAddress, err := s.app.CSRKeeper.DeployContract(s.ctx, csrSmartContract, &turnstileAddress)
 			s.Require().NoError(err)

--- a/x/csr/keeper/keeper_test.go
+++ b/x/csr/keeper/keeper_test.go
@@ -54,7 +54,6 @@ var s *KeeperTestSuite
 
 func TestKeeperTestSuite(t *testing.T) {
 	s = new(KeeperTestSuite)
-
 	suite.Run(t, s)
 
 	// Run Ginkgo integration tests

--- a/x/csr/module.go
+++ b/x/csr/module.go
@@ -169,14 +169,16 @@ func (AppModule) ConsensusVersion() uint64 { return 2 }
 func (am AppModule) BeginBlock(ctx sdk.Context, _ abci.RequestBeginBlock) {
 	// check in begin block whether the Turnstile has been deployed, if not, deploy it and set it to state
 	if _, found := am.keeper.GetTurnstile(ctx); !found {
-		// deploy turnstile
-		turnstile, err := am.keeper.DeployTurnstile(ctx)
-		// panic on errors, (Turnstile existence is invariant)
-		if err != nil {
-			panic(err)
+		if am.keeper.GetParams(ctx).EnableCsr { // only deploy if csr is enabled
+			// deploy turnstile
+			turnstile, err := am.keeper.DeployTurnstile(ctx)
+			// panic on errors, (Turnstile existence is invariant)
+			if err != nil {
+				panic(err)
+			}
+			// set the Turnstile address to state
+			am.keeper.SetTurnstile(ctx, turnstile)
 		}
-		// set the Turnstile address to state
-		am.keeper.SetTurnstile(ctx, turnstile)
 	}
 }
 

--- a/x/epochs/keeper/keeper_test.go
+++ b/x/epochs/keeper/keeper_test.go
@@ -1,6 +1,7 @@
 package keeper_test
 
 import (
+	"math/big"
 	"testing"
 	"time"
 
@@ -17,6 +18,12 @@ import (
 	tmversion "github.com/tendermint/tendermint/proto/tendermint/version"
 	"github.com/tendermint/tendermint/version"
 
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/evmos/ethermint/crypto/ethsecp256k1"
 	evm "github.com/evmos/ethermint/x/evm/types"
 
 	"github.com/Canto-Network/Canto/v6/app"
@@ -33,6 +40,12 @@ type KeeperTestSuite struct {
 	queryClientEvm evm.QueryClient
 	queryClient    types.QueryClient
 	consAddress    sdk.ConsAddress
+
+	address   common.Address
+	validator stakingtypes.Validator
+	ethSigner ethtypes.Signer
+
+	denom string
 }
 
 var s *KeeperTestSuite
@@ -56,6 +69,17 @@ func (suite *KeeperTestSuite) DoSetupTest(t require.TestingT) {
 
 	// init app
 	suite.app = app.Setup(checkTx, nil)
+
+	priv, err := ethsecp256k1.GenerateKey()
+	require.NoError(t, err)
+
+	suite.address = common.BytesToAddress(priv.PubKey().Address().Bytes())
+	suite.denom = "acanto"
+
+	// consensus key
+	privCons, err := ethsecp256k1.GenerateKey()
+	require.NoError(t, err)
+	suite.consAddress = sdk.ConsAddress(privCons.PubKey().Address())
 
 	// setup context
 	suite.ctx = suite.app.BaseApp.NewContext(checkTx, tmproto.Header{
@@ -87,6 +111,33 @@ func (suite *KeeperTestSuite) DoSetupTest(t require.TestingT) {
 	queryHelper := baseapp.NewQueryServerTestHelper(suite.ctx, suite.app.InterfaceRegistry())
 	types.RegisterQueryServer(queryHelper, suite.app.EpochsKeeper)
 	suite.queryClient = types.NewQueryClient(queryHelper)
+
+	bigInt := &big.Int{}
+	bigInt.SetUint64(100)
+	s.app.FeeMarketKeeper.SetBaseFee(suite.ctx, bigInt)
+
+	evmParams := suite.app.EvmKeeper.GetParams(suite.ctx)
+	evmParams.EvmDenom = suite.denom
+	suite.app.EvmKeeper.SetParams(suite.ctx, evmParams)
+
+	stakingParams := suite.app.StakingKeeper.GetParams(suite.ctx)
+	stakingParams.BondDenom = suite.denom
+	suite.app.StakingKeeper.SetParams(suite.ctx, stakingParams)
+
+	// Set Validator
+	valAddr := sdk.ValAddress(suite.address.Bytes())
+	validator, err := stakingtypes.NewValidator(valAddr, privCons.PubKey(), stakingtypes.Description{})
+	require.NoError(t, err)
+
+	validator = stakingkeeper.TestingUpdateValidator(suite.app.StakingKeeper, suite.ctx, validator, true)
+	suite.app.StakingKeeper.AfterValidatorCreated(suite.ctx, validator.GetOperator())
+	err = suite.app.StakingKeeper.SetValidatorByConsAddr(suite.ctx, validator)
+	require.NoError(t, err)
+
+	validators := s.app.StakingKeeper.GetValidators(s.ctx, 1)
+	suite.validator = validators[0]
+
+	suite.ethSigner = ethtypes.LatestSignerForChainID(s.app.EvmKeeper.ChainID())
 
 	// Set epoch start time and height for all epoch identifiers
 	identifiers := []string{types.WeekEpochID, types.DayEpochID}

--- a/x/inflation/keeper/keeper_test.go
+++ b/x/inflation/keeper/keeper_test.go
@@ -1,6 +1,7 @@
 package keeper_test
 
 import (
+	"math/big"
 	"testing"
 	"time"
 
@@ -17,11 +18,17 @@ import (
 	tmversion "github.com/tendermint/tendermint/proto/tendermint/version"
 	"github.com/tendermint/tendermint/version"
 
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/evmos/ethermint/crypto/ethsecp256k1"
 	evm "github.com/evmos/ethermint/x/evm/types"
 
 	"github.com/Canto-Network/Canto/v6/app"
 	epochstypes "github.com/Canto-Network/Canto/v6/x/epochs/types"
 	"github.com/Canto-Network/Canto/v6/x/inflation/types"
+
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
 )
 
 var denomMint = "acanto"
@@ -34,6 +41,9 @@ type KeeperTestSuite struct {
 	queryClientEvm evm.QueryClient
 	queryClient    types.QueryClient
 	consAddress    sdk.ConsAddress
+	address        common.Address
+	ethSigner      ethtypes.Signer
+	validator      stakingtypes.Validator
 }
 
 var s *KeeperTestSuite
@@ -58,6 +68,9 @@ func (suite *KeeperTestSuite) DoSetupTest(t require.TestingT) {
 	// init app
 	suite.app = app.Setup(checkTx, nil)
 
+	privCons, err := ethsecp256k1.GenerateKey()
+	require.NoError(t, err)
+	suite.consAddress = sdk.ConsAddress(privCons.PubKey().Address())
 	// setup context
 	suite.ctx = suite.app.BaseApp.NewContext(checkTx, tmproto.Header{
 		Height:          1,
@@ -88,6 +101,33 @@ func (suite *KeeperTestSuite) DoSetupTest(t require.TestingT) {
 	queryHelper := baseapp.NewQueryServerTestHelper(suite.ctx, suite.app.InterfaceRegistry())
 	types.RegisterQueryServer(queryHelper, suite.app.InflationKeeper)
 	suite.queryClient = types.NewQueryClient(queryHelper)
+
+	bigInt := &big.Int{}
+	bigInt.SetUint64(100)
+	s.app.FeeMarketKeeper.SetBaseFee(suite.ctx, bigInt)
+
+	evmParams := suite.app.EvmKeeper.GetParams(suite.ctx)
+	evmParams.EvmDenom = denomMint
+	suite.app.EvmKeeper.SetParams(suite.ctx, evmParams)
+
+	stakingParams := suite.app.StakingKeeper.GetParams(suite.ctx)
+	stakingParams.BondDenom = denomMint
+	suite.app.StakingKeeper.SetParams(suite.ctx, stakingParams)
+
+	// Set Validator
+	valAddr := sdk.ValAddress(suite.address.Bytes())
+	validator, err := stakingtypes.NewValidator(valAddr, privCons.PubKey(), stakingtypes.Description{})
+	require.NoError(t, err)
+
+	validator = stakingkeeper.TestingUpdateValidator(suite.app.StakingKeeper, suite.ctx, validator, true)
+	suite.app.StakingKeeper.AfterValidatorCreated(suite.ctx, validator.GetOperator())
+	err = suite.app.StakingKeeper.SetValidatorByConsAddr(suite.ctx, validator)
+	require.NoError(t, err)
+
+	validators := s.app.StakingKeeper.GetValidators(s.ctx, 1)
+	suite.validator = validators[0]
+
+	suite.ethSigner = ethtypes.LatestSignerForChainID(s.app.EvmKeeper.ChainID())
 
 	// Set epoch start time and height for all epoch identifiers
 	identifiers := []string{epochstypes.WeekEpochID, epochstypes.DayEpochID}

